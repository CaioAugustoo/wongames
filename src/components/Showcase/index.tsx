import GameCardSlider, { GameCardSliderProps } from 'components/GameCardSlider'
import Heading from 'components/Heading'
import { GameCardProps } from 'components/GameCard'
import Highlight, { HighLightProps } from 'components/Highlight'

import * as S from './styles'

export type ShowcaseProps = {
  title?: string
  highlight?: HighLightProps
  games?: GameCardProps[]
} & Pick<GameCardSliderProps, 'color'>

const Showcase = ({
  title,
  highlight,
  games,
  color = 'white'
}: ShowcaseProps) => (
  <S.Wrapper>
    {!!title && (
      <Heading lineLeft lineColor="secondary">
        {title}
      </Heading>
    )}
    {!!highlight && <Highlight {...highlight} />}
    {!!games && <GameCardSlider items={games} color={color} />}
  </S.Wrapper>
)

export default Showcase
