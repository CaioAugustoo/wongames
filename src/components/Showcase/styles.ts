import styled, { css } from 'styled-components'

import media from 'styled-media-query'

import * as HeadingStyles from 'components/Heading/styles'
import * as GameCardSliderStyles from 'components/GameCardSlider/styles'
import * as HighlightStyles from 'components/Highlight/styles'

import { Container } from 'components/Container'

export const Wrapper = styled(Container).attrs({ as: 'section' })`
  ${({ theme }) => css`
    ${HeadingStyles.Wrapper},
    ${HighlightStyles.Wrapper},
    ${GameCardSliderStyles.Wrapper} {
      margin-bottom: ${theme.spacings.medium};
    }
    ${HighlightStyles.Wrapper} {
      ${media.lessThan('medium')`
        margin-right: calc(-${theme.grid.gutter} / 2);
        margin-left: calc(-${theme.grid.gutter} / 2);
      `}
    }
    ${GameCardSliderStyles.Wrapper} {
      ${media.lessThan('huge')`
        margin-right: calc(-${theme.grid.gutter} / 2);
      `}
    }
    margin-bottom: calc(${theme.spacings.large} * 2);
  `}
`
