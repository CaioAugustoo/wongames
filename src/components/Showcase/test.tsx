import 'match-media-mock'
import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import gamesMock from 'components/GameCardSlider/mock'
import highlightMock from 'components/Highlight/mock'

import Showcase from '.'

const props = {
  title: 'Most Popular',
  highlight: highlightMock,
  games: gamesMock.slice(0, 1)
}

describe('<Showcase />', () => {
  it('should render full showcase', () => {
    renderWithTheme(<Showcase {...props} />)

    expect(
      screen.getByRole('heading', { name: /most popular/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: highlightMock.title })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: gamesMock[0].title })
    ).toBeInTheDocument()
  })

  it('should render without title', () => {
    renderWithTheme(
      <Showcase games={props.games} highlight={props.highlight} />
    )

    screen.getByRole('heading', { name: highlightMock.title })
    screen.getByRole('heading', { name: gamesMock[0].title })

    expect(
      screen.queryByRole('heading', { name: /most popular/i })
    ).not.toBeInTheDocument()
  })

  it('should render without highlight', () => {
    renderWithTheme(<Showcase title={props.title} games={props.games} />)

    screen.getByRole('heading', { name: /most popular/i })
    screen.getByRole('heading', { name: gamesMock[0].title })

    expect(
      screen.queryByRole('heading', { name: highlightMock.title })
    ).not.toBeInTheDocument()
  })

  it('should render without games', () => {
    renderWithTheme(
      <Showcase title={props.title} highlight={props.highlight} />
    )

    screen.getByRole('heading', { name: /most popular/i })
    screen.getByRole('heading', { name: highlightMock.title })

    expect(
      screen.queryByRole('heading', { name: gamesMock[0].title })
    ).not.toBeInTheDocument()
  })
})
