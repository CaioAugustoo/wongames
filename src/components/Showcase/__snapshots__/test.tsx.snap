// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Showcase /> should render full showcase 1`] = `
.c21 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c10 {
  background: #47c8ff;
  color: #FAFAFA;
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c10:hover {
  background: #2499ae;
}

.c25 {
  background: #47c8ff;
  color: #FAFAFA;
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  height: 3rem;
  font-size: 1.2rem;
}

.c25:hover {
  background: #2499ae;
}

.c25 svg {
  width: 1.5rem;
}

.c25 svg + span {
  margin-left: 0.8rem;
}

.c14 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  background-color: #FAFAFA;
}

.c15 {
  height: 14rem;
  width: 100%;
  background: #f6f7f8;
  background-image: linear-gradient( to right,#f6f7f8 0%,#edeef1 20%,#f6f7f8 40%,#f6f7f8 100% );
  background-size: 80rem 14rem;
  -webkit-animation: placeholderShimmer 1s linear infinite forwards;
  animation: placeholderShimmer 1s linear infinite forwards;
}

.c15 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  height: 100%;
  margin: 1.6rem;
}

.c17 {
  max-width: calc(100% - 2.5rem);
}

.c18 {
  font-size: 1.6rem;
  line-height: 1.6rem;
  font-weight: 600;
  color: #030517;
}

.c19 {
  font-size: 1.4rem;
  font-weight: 600;
  color: #8F8F8F;
}

.c20 {
  color: #2499ae;
  position: absolute;
  right: 0;
  top: -0.5rem;
  cursor: pointer;
}

.c20 svg {
  width: 2.5rem;
}

.c22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-top: 0.8rem;
}

.c22 button {
  background-color: #f062c0;
}

.c22 button:hover {
  background-color: #f062c0;
  opacity: 0.9;
}

.c23 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 3rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #8F8F8F;
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
  margin-right: 0.8rem;
}

.c24 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 3rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FAFAFA;
  padding: 0 0.8rem;
  background-color: #47C8FF;
  border-radius: 0.4rem;
  margin-right: calc(0.8rem / 2);
}

.c13 .slick-slider {
  position: relative;
  display: block;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-touch-callout: none;
  -khtml-user-select: none;
  -ms-touch-action: pan-y;
  touch-action: pan-y;
  -webkit-tap-highlight-color: transparent;
}

.c13 .slick-list {
  position: relative;
  display: block;
  overflow: hidden;
  margin: 0;
  padding: 0;
}

.c13 .slick-list:focus {
  outline: none;
}

.c13 .slick-list.dragging {
  cursor: pointer;
  cursor: hand;
}

.c13 .slick-slider .slick-track,
.c13 .slick-slider .slick-list {
  -webkit-transform: translate3d(0,0,0);
  -moz-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  -o-transform: translate3d(0,0,0);
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
}

.c13 .slick-track {
  position: relative;
  top: 0;
  left: 0;
  display: block;
}

.c13 .slick-track:before,
.c13 .slick-track:after {
  display: table;
  content: '';
}

.c13 .slick-track:after {
  clear: both;
}

.c13 .slick-loading .slick-track {
  visibility: hidden;
}

.c13 .slick-slide {
  display: none;
  float: left;
  height: 100%;
  min-height: 1px;
}

.c13 [dir='rtl'] .slick-slide {
  float: right;
}

.c13 .slick-slide img {
  display: block;
}

.c13 .slick-slide.slick-loading img {
  display: none;
}

.c13 .slick-slide.dragging img {
  pointer-events: none;
}

.c13 .slick-initialized .slick-slide {
  display: block;
}

.c13 .slick-loading .slick-slide {
  visibility: hidden;
}

.c13 .slick-vertical .slick-slide {
  display: block;
  height: auto;
  border: 1px solid transparent;
}

.c13 .slick-arrow.slick-hidden {
  display: none;
}

.c12 .slick-track,
.c12 .slick-list {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 .slick-slide > div {
  margin: 0 0.8rem;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  height: 100%;
}

.c12 .slick-list {
  margin: 0 -0.8rem;
}

.c12 .slick-prev,
.c12 .slick-next {
  display: block;
  color: #FAFAFA;
  cursor: pointer;
  position: absolute;
  top: 50%;
  width: 2.5rem;
  height: 2.5rem;
  padding: 0;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
}

.c12 .slick-prev {
  left: -5.6rem;
}

.c12 .slick-next {
  right: -5.6rem;
}

.c12 .slick-prev.slick-disabled,
.c12 .slick-next.slick-disabled {
  visibility: hidden;
}

.c3 {
  color: #FAFAFA;
  padding-left: 0.8rem;
  border-left: 0.7rem solid #47C8FF;
  font-size: 2.0rem;
}

.c5 {
  position: relative;
  height: 23rem;
  display: grid;
  background-image: url(/img/red-dead-img.jpg);
  background-position: center center;
  background-size: cover;
  grid-template-areas: 'floatimage content';
}

.c5::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.6);
}

.c5 .c6 {
  text-align: right;
}

.c7 {
  grid-area: content;
  z-index: 10;
  padding: 1.6rem;
}

.c8 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c9 {
  font-size: 1.4rem;
  font-weight: 300;
  color: #FAFAFA;
  margin-bottom: 3.2rem;
}

.c0 {
  width: 100%;
  max-width: 130rem;
  margin-right: auto;
  margin-left: auto;
  padding-left: calc(3.2rem / 2);
  padding-right: calc(3.2rem / 2);
}

.c1 {
  margin-bottom: calc(4.0rem * 2);
}

.c1 .c2,
.c1 .c4,
.c1 .c11 {
  margin-bottom: 3.2rem;
}

@media (max-width:1440px) {
  .c12 {
    overflow-x: hidden;
  }
}

@media (min-width:1170px) {
  .c12 .slick-slide > div {
    margin: 0 1.6rem;
  }

  .c12 .slick-list {
    margin: 0 -1.6rem;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c5 {
    height: 32rem;
  }
}

@media (min-width:768px) {
  .c7 {
    -webkit-align-self: end;
    -ms-flex-item-align: end;
    align-self: end;
    padding: 4.0rem;
  }
}

@media (min-width:768px) {
  .c8 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c9 {
    font-size: 1.8rem;
  }
}

@media (max-width:768px) {
  .c1 .c4 {
    margin-right: calc(-3.2rem / 2);
    margin-left: calc(-3.2rem / 2);
  }
}

@media (max-width:1440px) {
  .c1 .c11 {
    margin-right: calc(-3.2rem / 2);
  }
}

<section
  class="c0 c1"
>
  <h2
    class="c2 c3"
    color="white"
  >
    Most Popular
  </h2>
  <section
    class="c4 c5"
  >
    <div
      class="c6 c7"
    >
      <h2
        class="c8"
      >
        Read Dead is back!
      </h2>
      <h3
        class="c9"
      >
        Come see Johnâ€™s new adventures
      </h3>
      <a
        class="c10"
        href="/games/rdr2"
      >
        <span>
          Buy now
        </span>
      </a>
    </div>
  </section>
  <section
    class="c11 c12"
    color="white"
  >
    <section
      class="c13"
    >
      <div
        class="slick-slider slick-initialized"
      >
        
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <article
                  class="c14"
                >
                  <div
                    class="c15"
                  >
                    <img
                      alt="Population Zero"
                      src="https://source.unsplash.com/user/willianjusten/300x140"
                    />
                  </div>
                  <div
                    class="c16"
                  >
                    <div
                      class="c17"
                    >
                      <h3
                        class="c18"
                      >
                        Population Zero
                      </h3>
                      <h4
                        class="c19"
                      >
                        Rockstar Games
                      </h4>
                    </div>
                    <div
                      class="c20"
                      role="button"
                    >
                      <svg
                        aria-hidden="true"
                        aria-label="Adicionar a lista de desejos"
                        class="c21"
                        fill="currentColor"
                        focusable="false"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M0 0h24v24H0V0z"
                          fill="none"
                        />
                        <path
                          d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                        />
                      </svg>
                    </div>
                    <div
                      class="c22"
                    >
                      <div
                        class="c23"
                      >
                        R$ 235,00
                      </div>
                      <div
                        class="c24"
                      >
                        R$ 215,00
                      </div>
                      <button
                        class="c25"
                      >
                        <svg
                          aria-hidden="true"
                          class="c21"
                          fill="currentColor"
                          focusable="false"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 0h24v24H0V0z"
                            fill="none"
                          />
                          <path
                            d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4l-3.87 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2z"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </article>
              </div>
            </div>
          </div>
        </div>
        
        
      </div>
    </section>
  </section>
</section>
`;
